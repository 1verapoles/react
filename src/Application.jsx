import React from 'react'
import {
  Switch,
  Route,
  useLocation
} from "react-router-dom";
import { CSSTransition, TransitionGroup } from 'react-transition-group'
import './index.css'
import App from './App'
import Error from './Error'
import Header from './Header'
import Details from './Details'
import About from './About'



function Application() {
  let location = useLocation();
  return (<>
    <Header />
    <TransitionGroup>
      <CSSTransition timeout={2000} classNames="page" key={location.key}>
        <Switch location={location} >
          <Route exact path="/">
            <App />
          </Route>
          <Route path="/about">
            <About />
          </Route>
          <Route path="/details/:id">
            <Details />
          </Route>
          <Route path="*">
            <Error />
          </Route>
        </Switch>
      </CSSTransition>
    </TransitionGroup>
  </>)
}


export default Application