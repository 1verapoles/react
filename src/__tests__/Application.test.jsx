import React from 'react';
import { BrowserRouter } from "react-router-dom";
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../store';
import Application from '../Application';

describe('Testing Application component', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<Provider store={store}><BrowserRouter><Application /></BrowserRouter></Provider>);

    expect(asFragment()).toMatchSnapshot();
  });
});
