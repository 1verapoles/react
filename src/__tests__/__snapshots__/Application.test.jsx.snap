// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Application component renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light bg-light"
    >
      <button
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-target="#navbarNav"
        data-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          class="navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/about"
            >
              About
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <div>
    <div
      class="container"
    >
      <div
        class="form-group form"
      >
        <label
          class="d-flex justify-content-center"
          for="search"
        >
          Введите поисковую фразу:
        </label>
        <input
          aria-describedby="search"
          class="form-control"
          id="search"
          type="text"
          value=""
        />
      </div>
      <div
        class="d-flex justify-content-center"
      >
        Sort by:
      </div>
      <div
        class="d-flex justify-content-center my-2"
      >
        <div
          class="form-check form-check-inline"
        >
          <input
            class="form-check-input"
            id="inlineRadio1"
            name="inlineRadioOptions"
            type="radio"
            value="relevancy"
          />
          <label
            class="form-check-label"
            for="inlineRadio1"
          >
            relevancy
          </label>
        </div>
        <div
          class="form-check form-check-inline"
        >
          <input
            class="form-check-input"
            data-testid="popl"
            id="inlineRadio2"
            name="inlineRadioOptions"
            type="radio"
            value="popularity"
          />
          <label
            class="form-check-label"
            for="inlineRadio2"
          >
            popularity
          </label>
        </div>
        <div
          class="form-check form-check-inline"
        >
          <input
            class="form-check-input"
            id="inlineRadio3"
            name="inlineRadioOptions"
            type="radio"
            value="publishedAt"
          />
          <label
            class="form-check-label"
            for="inlineRadio3"
          >
            publishedAt
          </label>
        </div>
      </div>
      <div
        class="d-flex justify-content-center my-2"
      >
        <div
          class="form-row"
        >
          <div
            class="form-group col-md-6"
          >
            <label
              for="pageSize"
            >
              Кол-во статей на странице
            </label>
            <input
              class="form-control"
              data-testid="art"
              id="pageSize"
              min="1"
              placeholder="20"
              type="number"
              value=""
            />
          </div>
          <div
            class="form-group col-md-6"
          >
            <label
              for="pageNumber"
            >
              Номер страницы
            </label>
            <input
              class="form-control"
              id="pageNumber"
              min="1"
              placeholder="1"
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="d-flex justify-content-center my-2 mb-5"
      >
        Количество страниц: 
      </div>
      <div
        class="d-flex flex-wrap"
      />
    </div>
  </div>
</DocumentFragment>
`;
